src/test/java/org/cactoos/iterable/MappedWithIndexTest.java

```diff
@@ -0,0 +1,108 @@
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterable;
+
+import java.util.Collections;
+import org.cactoos.text.Joined;
+import org.cactoos.text.TextOf;
+import org.cactoos.text.Upper;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+
+/**
+ * Tests for {@link MappedWithIndex}.
+ * @since 1.0.0
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+final class MappedWithIndexTest {
+    @Test
+    void transformsList() throws Exception {
+        new Assertion<>(
+            "must transform an iterable",
+            new MappedWithIndex<>(
+                (input, index) -> new Upper(
+                    new Joined(
+                        "-",
+                        new TextOf(index.toString()),
+                        new TextOf(input)
+                    )
+                ),
+                new IterableOf<>(
+                    "hello", "world", "друг"
+                )
+            ).iterator().next().asString(),
+            new IsEqual<>("0-HELLO")
+        ).affirm();
+    }
+
+    @Test
+    void transformsEmptyList() {
+        new Assertion<>(
+            "must transform an empty iterable",
+            new MappedWithIndex<>(
+                (input, index) -> {
+                    Assertions.fail("must do not be executed");
+                    return input;
+                },
+                Collections.emptyList()
+            ),
+            Matchers.emptyIterable()
+        ).affirm();
+    }
+
+    @Test
+    void string() {
+        new Assertion<>(
+            "must convert to string",
+            new MappedWithIndex<>(
+                (x, index) -> x * index * 2,
+                new IterableOf<>(1, 2, 3)
+            ).toString(),
+            Matchers.equalTo("0, 4, 12")
```

Comment:
@dearrudam please avoid using any static method, here you can use `IsEqual`, and in `transformsEmptyList` you can directly throw an exception and also use new IterableOf<>() instead of `Collections.emptyList()`,  etc. You can also replace matchers such as `Matchers.emptyIterable()` with their equivalent objects instantiation with `new` (o use `IsEqual`). If some are unnice to use, don't hesitate to open new issues to cactoos-matchers to improve OO matchers :)

Reply:
@victornoel I'll fix it! Thanks!